tf_encrypted/protocol/pond/pond.py

```diff
@@ -2136,9 +2193,12 @@ def __init__(self, prot, variable0, variable1, is_scaled):
     )
     self.variable0 = variable0
     self.variable1 = variable1
-    self.initializer = tf.group(
-        *[var.initializer for var in [variable0, variable1]]
-    )
+
+    # TensorFlow 2.0 variables don't have initializers??
```

Comment:
what's the deal here? it looks like the initializer is still there, but automatically invoked in eager and graph mode

Reply:
In eager and with tf.functions the initializers aren't stored on the variables because they are `ResourceVariables`. With eager disabled and using a session they are stored here. Since we're still using sessions in some places we needed this check to take advantage of the initializers.